### PR TITLE
Prefs page with forked nassh prefs editor.

### PR DESCRIPTION
**Pull request by mjkelly**
This change imports nassh_preferences_editor.js from Chromium libapps,
with a few changes to mock out some i18n methods and force the
profile-selection code to pick the 'mosh' profile.

### CODE REVIEW

**Comment by rpwoodbu** [comment]
Sorry for the very long delay getting to this. I very much appreciate you helping out.

I'm slightly reluctant to fork the prefs page from nassh if we can help it. Is it possible that they've folded that into hterm by now so we can just reference it?

**Comment by mjkelly** [comment]
As of a few weeks ago, they haven't. Robert said he'd accept a CL that moved the non-display logic of the prefs page into hterm, but he had no plans to do it.

If you really don't like forking the prefs code, I can have a look at moving the nassh prefs code into hterm, but may be a few months before I look at it.

File: src/app/mosh_client.html [hunk 1]
```diff
@@ -63,6 +63,7 @@
     <td align="right" valign="top">
       <a href="https://github.com/rpwoodbu/mosh-chrome/wiki/Documentation" target="_blank">Help</a><br>
       <a href="" id="ssh-key">Add ssh key</a><br>
+      <a href="" id="prefs">Terminal preferences</a><br>
```
Comment (rpwoodbu) [inline src/app/mosh_client.html]: How about just "Preferences"? We could put other things in there in the future that aren't hterm-related. I don't want to make tracking upstream changes too difficult, though, but I think we can manage.

File: src/app/mosh_client.js [hunk 2]
```diff
@@ -157,3 +159,18 @@ function onSshKeyClick(e) {
   // Prevent default handling.
   return true;
 }
+
+function onPrefsClick(e) {
+      //'nassh_preferences_editor.html',
```
Comment (rpwoodbu) [inline src/app/mosh_client.js]: Spurious comment?

File: src/app/mosh_prefs.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
+<html>
```
Comment (rpwoodbu) [inline src/app/mosh_prefs.html]: Insert copyright notice. Cargo-cult from mosh_client.html.

File: src/app/mosh_prefs.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        vertical-align: middle;
+      }
+      span.profile-ui input[type='text'],
+      span.profile-ui input[type='number'],
+      span.profile-ui textarea {
+        width: 100%;
+      }
+
+      /* Instantiating a terminal in this window means that we must manually
+      create a scrollable area. */
```
Comment (rpwoodbu) [inline src/app/mosh_prefs.html]: Prefer "//" style comments.

**Comment by rpwoodbu** [comment]
I can live with the forking. We can try to work toward eliminating it in the future perhaps.

File: src/app/mosh_prefs.html (hunk 4 above)
Comment (mjkelly) [inline src/app/mosh_prefs.html]: It's CSS, unfortunately. :-/

File: src/app/mosh_prefs.html (hunk 3 above)
Comment (mjkelly) [inline src/app/mosh_prefs.html]: I wrote the HTML and took the CSS from nassh. I'll split them into separate files so the distinction is clean.

**Comment by mjkelly** [comment]
I'm hoping to push a new version that addresses these comments this week. (Really I swear I haven't forgotten about this!)

File: src/app/mosh_client.html (hunk 1 above)
Comment (mjkelly) [inline src/app/mosh_client.html]: Done.

File: src/app/mosh_client.js (hunk 2 above)
Comment (mjkelly) [inline src/app/mosh_client.js]: Removed.

**Comment by mjkelly** [comment]
PTAL, I've updated the pull request.

I've split the prefs page into mosh_prefs.html (all new) and nassh_preferences.css, which contains CSS from nassh. I hope the status of each of the files is clear now.

(I've squashed intermediate commits and overwritten history on my pull request branch, which someone on stackoverflow said was the thing to do. I'm shamefully ignorant about github pull request procedures, so let me know if I broke anything.)

**Comment by rpwoodbu** [comment]
I have _finally_ gotten to this! Thanks so much for doing it. Your change (with a rebase) is in SHA f588d7b109fdb329c5e21bfa45ec300b3b499684. I had to fix a tiny but serious bug in SHA c8073e94a06ea18499823f3a9dd2e0b8e82a6fcb (maybe the browser became more strict); PTAL to make sure that looks right.

This is rolling out on the dev channel right now.

**Pull request closed without merging**